osidb/tests/test_e2e.py

```diff
@@ -644,3 +645,188 @@ def test_cveorg_with_workflows(
+            "comment_zero": "This is a test for embargoed ACL validation",
+            "impact": "CRITICAL",
+            "components": ["kernel"],
+            "source": "REDHAT",
+            "reported_dt": "2024-08-06T00:00:00.000Z",
+            "unembargo_dt": "2024-08-07T00:00:00.000Z",
+            "embargoed": True,
+        }
+
+        response = auth_client().post(
+            f"{test_api_uri}/flaws",
+            embargoed_flaw_data,
+            format="json",
+            HTTP_BUGZILLA_API_KEY=bugzilla_token,
+            HTTP_JIRA_API_KEY=jira_token,
+        )
+
+        # 2.1) Validate embargoed flaw was created successfully
+        assert response.status_code == 201
+        embargoed_body = response.json()
+        embargoed_flaw_uuid = embargoed_body["uuid"]
+
+        # 2.2) Get the created embargoed flaw from database to check ACLs
+        embargoed_flaw = Flaw.objects.get(uuid=embargoed_flaw_uuid)
+
+        # 2.3) Verify ACL values are set correctly for embargoed flaw
+        assert embargoed_flaw.is_embargoed
+        assert not embargoed_flaw.is_public
+        assert not embargoed_flaw.is_internal
+
+        # Check that ACLs match expected embargoed groups
+        expected_embargoed_read_acls = [
+            uuid.UUID(acl) for acl in generate_acls([settings.EMBARGO_READ_GROUP])
+        ]
+        expected_embargoed_write_acls = [
+            uuid.UUID(acl) for acl in generate_acls([settings.EMBARGO_WRITE_GROUP])
+        ]
+
+        assert embargoed_flaw.acl_read == expected_embargoed_read_acls
+        assert embargoed_flaw.acl_write == expected_embargoed_write_acls
+
+        # 3) Test access control based on ACLs
+        # 3.1) Internal flaw should be inaccessible to unauthenticated client
+        response = client.get(f"{test_api_uri}/flaws/{flaw_uuid}")
+        assert response.status_code == 404
+
+        # 3.2) Embargoed flaw should NOT be accessible to unauthenticated client
+        response = client.get(f"{test_api_uri}/flaws/{embargoed_flaw_uuid}")
+        assert response.status_code == 404
+
+        # 3.3) Authenticated client should be able to access both flaws
+        response = auth_client().get(f"{test_api_uri}/flaws/{flaw_uuid}")
+        assert response.status_code == 200
+
+        response = auth_client().get(f"{test_api_uri}/flaws/{embargoed_flaw_uuid}")
+        assert response.status_code == 200
+
+        # 4) Test ACL modification via REST API
+        # 4.1) Update the internal flaw to change its embargo status
+        updated_flaw_data = {
+            "title": "Updated Test ACL values",
+            "comment_zero": "This is an updated test for ACL validation",
+            "impact": "IMPORTANT",
+            "components": ["curl"],
+            "source": "REDHAT",
+            "reported_dt": "2024-08-06T00:00:00.000Z",
+            "unembargo_dt": "3339-08-06T00:00:00.000Z",
+            "updated_dt": flaw.updated_dt,
+            "embargoed": True,  # Change to embargoed
```

Comment:
Changing the visibility to something more strict does not make much sense and ideally/hopefully would be prevented by the validation. The reason is that when something was already visible publicly or internally, it is can be already cached somewhere and trying to hide it is already futile.

Reply:
I wondered if it didn't make sense, but then wondered if there are ever edge cases where something needs to become embargoed, but upon looking at it again, a user wouldn't send this type of payload (`"embargoed": True`) with a PUT request. Should I strengthen the validation? This didn't trigger any complaint

Reply:
So there actually is that validation https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/models/flaw/flaw.py#L691 but again only meant for the binary embargo/public world so does not get triggered by internal-to-embargoed change. You may generalize it so it works for any late attempt to restrict the access.